### PR TITLE
🌎Added WorldHideaway Presence

### LINKED ISSUE #3101: WorldHideAway | worldhideaway.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
srjoel#0251

**What's the name of the service?** 
www.worldhideaway.com

**What should the Presence display?** 
The different sections of web (Guides section, news, home)

**If possible, please provide a logo for the service (512x512 minimum)** 
Sin título-1_Mesa de trabajo 1

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #3101 
Added Full Website Functionality.
wh1
wh2
wh3
wh4
wh5
wh6

### CODE REVIEW

File: websites/W/Worldhideaway/presence.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  page = window.location.pathname,
+  expage = window.location.href,
+  host = document.location.hostname; 
+
+    presenceData.startTimestamp = browsingStamp;
+    if (host === "www.worldhideaway.com" || host === "worldhideaway.com") {
```
Comment (Bas950) [inline websites/W/Worldhideaway/presence.ts:16]: useless if

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/W/Worldhideaway/presence.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  }
+  const check = document.querySelector("#main > div > div > section.elementor-section.elementor-top-section.elementor-element.elementor-element-7cf6a1d.elementor-section-boxed.elementor-section-height-default.elementor-section-height-default > div > div > div > div > div > div > div > h1");
+   if (check) {
+    presenceData.details = "Leyendo:";
+    presenceData.state = check.textContent;
+    presenceData.smallImageKey = "reading";
+   }
+
+  else if (presenceData.details === null) {
+    presence.setTrayTitle();
```
Comment (Bas950) [inline websites/W/Worldhideaway/presence.ts:108]: should this be an else if?

File: websites/W/Worldhideaway/presence.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+title: HTMLElement;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  page = window.location.pathname,
+  expage = window.location.href;
+
+    presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/W/Worldhideaway/presence.ts:14]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "logo",
    startTimestamp: browsingStamp
  },
  page = window.location.pathname,
  expage = window.location.href;
```

File: websites/W/Worldhideaway/presence.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      search = document.querySelector("#masthead > div.header-search-wrap > div > form > label > input");
+      if (!search) {
+        presenceData.details = "Viendo Inicio";
+      } else if (search.value !== "") {
+        presenceData.details = "Buscando:";
+        presenceData.state = search.value;
+        presenceData.smallImageKey = "searching";
+      } else { 
+        presenceData.details = "Viendo Inicio";
+      }
```
Comment (ririxidev) [inline websites/W/Worldhideaway/presence.ts:25]: what? XD
```suggestion
      if (!search || search.value === "") {
        presenceData.details = "Viendo Inicio";
      } else {
        presenceData.details = "Buscando:";
        presenceData.state = search.value;
        presenceData.smallImageKey = "searching";
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @DarkVillager!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Comment by ririxidev** [comment]
Lol i forgot ``}`` in last ``if`` fix it.

**Review by ririxidev — APPROVED** [review]
K

**Review by StrikerFRFX — APPROVED** [review]
lgtm

**Pull request merged**